templates/demoEC2.yml

```
@@ -0,0 +1,97 @@
+AWSTemplateFormatVersion: "2010-09-09"
+Resources: 
+  demoEC2InstanceProfile:
+    Type: AWS::IAM::InstanceProfile
+    Properties: 
+      InstanceProfileName: demo-ec2-instance-profile
+      Path: /
+      Roles: 
+       - !Ref demoEC2InstanceRole
+
+  demoEC2InstanceRole:
+    Type: AWS::IAM::Role
+    Properties:
+      RoleName: demo-ec2-instance-role
+      AssumeRolePolicyDocument:
+        Version: 2012-10-17
+        Statement:
+          -
+            Effect: Allow
+            Principal:
+              Service:
+                - ec2.amazonaws.com
+            Action:
+              - sts:AssumeRole
+      Path: /
+      ManagedPolicyArns: 
+      - arn:aws:iam::aws:policy/AmazonDynamoDBFullAccess
+      - arn:aws:iam::aws:policy/AmazonSQSFullAccess
+
+  demoWebSecurityGroup:
+    Type: 'AWS::EC2::SecurityGroup'
+    Properties:
+      GroupDescription: "Web Security Group"
+      SecurityGroupIngress:
+      - IpProtocol: tcp
+        FromPort: 80
+        ToPort: 80
+        CidrIp: 0.0.0.0/0
+
+  demoSSHSecurityGroup:
+    Type: 'AWS::EC2::SecurityGroup'
+    Properties:
+      GroupDescription: "SSH Security Group"
+      SecurityGroupIngress:
+      - IpProtocol: tcp
+        FromPort: 22
+        ToPort: 22
+        CidrIp: 0.0.0.0/0
+
+  demoEC2Instance: 
+    Type: 'AWS::EC2::Instance'
+    DependsOn: 
+    - "demoWebSecurityGroup"
+    - "demoSSHSecurityGroup"
+    - "demoEC2InstanceProfile"
+    Properties: 
+      ImageId: "ami-00399ec92321828f5"
+      KeyName: "demo"
+      InstanceType: "t2.micro"
+      SecurityGroups:
+      - Ref: "demoWebSecurityGroup"
```

You can use short form syntax here:

```suggestion
      - !Ref demoWebSecurityGroup
```

---

Done